common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1291,16 +1291,17 @@ def _add_timezone_string(self, date_time):
     def forum_posts_allowed(self):
         date_proxy = Date()
         try:
-            blackout_periods = [(date_proxy.from_json(start),
-                                 date_proxy.from_json(end))
-                                for start, end
-                                in self.discussion_blackouts]
+            blackout_periods = [(date_proxy.from_json(blackout[0]),
+                                 date_proxy.from_json(blackout[1]))
+                                for blackout
+                                in self.discussion_blackouts
+                                if blackout]
```

Comment:
You can do this thing by filtering the empty tuples first (https://docs.python.org/2/library/functions.html#filter), this will keep the understandable names of elements (start, end).
```python
[(date_proxy.from_json(start),
  date_proxy.from_json(end)
  for start, end
  in filter(None, self.discussion_blackouts))]
```